base_decorator.rb

```diff
@@ -0,0 +1,14 @@
+require './nameable'
+
+class Decorator < Nameable
+  attr_accessor :nameable
+
+  def initialize(nameable)
+    super
```

Comment:
- [ ] Please be extra careful using the super keyword, the default behavior is to call the same method on the parent with the same arguments, but... What if the methods don't share the same number of arguments? 😱 In this case, Nameable doesn't have an explicit initialize method, so by default, it receives no arguments, but here you're calling it with `nameable` as an argument which causes an error, you can read more about the `super` keyword [here](https://www.rubyguides.com/2018/09/ruby-super-keyword/). Please consider adding a `()` to indicate that it doesn't receive any arguments, like this: 🤓 🔧 

```suggestion
    super()
```